src/JLD.jl

```diff
@@ -203,7 +203,7 @@ function jldopen(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Boo
             if sz < 512
                 error("File size indicates $filename cannot be a Julia data file")
             end
-            magic = Array(UInt8, 512)
+            magic = Array{UInt8}(512)
```

Comment:
Done for the obvious `Vector`/`Matrix` cases. 

There are still a couple of functions floating around with the argument `dims::@compat Tuple{Vararg{Int}}`, which I could rewrite as `dims::NTuple{N,Int}` so that I could use `N` in the associated array constructors. I held off as I wasn't sure if it was necessary.